proto/dataplane.proto

```diff
@@ -29,72 +29,101 @@ enum Error {
   ERROR_UNKNOWN_ERROR = 3;
 }
 
-enum IfType {
-  IF_TYPE_UPLINK = 0;
-  IF_TYPE_MANAGEMENT = 1;
-  IF_TYPE_VXLAN = 2;
-}
-
 message GetConfigGenerationRequest {
 }
 
 message GetConfigGenerationResponse {
   uint64 generation = 1;
 }
 
-message Subnet {
-  string cidr = 1;
-  string name = 2;
+
+/* ================================================ */
+/* ==== Interfaces (both underlat and overlay) ==== */
+/* ================================================ */
+
+message Interface {
+  string name = 1;
+  string ipaddr = 2;
+  IfType type = 3;
+  IfRole role = 4;
 }
 
-message VPC {
-  string id = 1;
-  string name = 2;
-  uint32 vni = 3;
-  repeated Subnet subnets = 4;
+enum IfType {
+  IF_TYPE_ETHERNET = 0;
+  IF_TYPE_VLAN = 1;
+  IF_TYPE_LOOPBACK = 2;
+  IF_TYPE_VTEP = 3;
 }
 
-message Device {
-  uint32 index = 1;
-  string ipaddr = 2;
-  string name = 3;
-  string pciaddr = 4;
-  IfType type = 5;
+enum IfRole {
+  IF_ROLE_FABRIC = 0;
+  IF_ROLE_MANAGEMENT = 1;
+  IF_ROLE_EXTERNAL = 2;
 }
 
-message PeeringIPs {
+/* ================= */
+/* ==== Overlay ==== */
+/* ================= */
+
+message VpcSubnetAs {
```

Comment:
I think it's a bit misleading name as it's just a one of the IPs/Nots for the peering, not a VPC subnet